File: Medium/1727. Largest Submatrix With Rearrangements.java
class Solution {
    public int largestSubmatrix(int[][] matrix) {
        int R = matrix.length, C = matrix[0].length;
        for (int c = 0; c < C; c++) {
            for (int r = 1; r < R; r++) {
                if (matrix[r][c] == 0) {
                    continue;
                }
                matrix[r][c] += matrix[r - 1][c];
            }
        }
        int ans = 0;
        for (int[] row : matrix) {
            Arrays.sort(row);
            for (int c = C - 1, l = 1; c > -1 && row[c] > 0; c--, l++) {
                ans = Math.max(row[c] * l, ans);
            }
        }
        return ans;
    }
}

File: Medium/1727. Largest Submatrix With Rearrangements.py
class Solution:
    def largestSubmatrix(self, matrix: List[List[int]]) -> int:
        R, C = len(matrix), len(matrix[0])
        for c in range(C):
            for r in range(1, R):
                if matrix[r][c]:
                    matrix[r][c] += matrix[r - 1][c]
        res = 0
        for row in matrix:
            for c, v in enumerate(sorted(row, reverse = True)):
                if not v:
                    break
                res = max(v * (c + 1), res)
        return res
